libosdp/src/channel.rs

```diff
@@ -38,6 +40,17 @@ impl From<std::io::Error> for ChannelError {
     }
 }
 
+#[cfg(not(feature = "std"))]
+impl<E: embedded_io::Error + Sized> From<E> for ChannelError {
+    fn from(value: E) -> Self {
+        match value.kind() {
+            //TODO determine if this is the correct error kind
+            // embedded_io::ErrorKind::TimedOut => ChannelError::WouldBlock,
```

Comment:
`embedded_io::ErrorKind::TimedOut` cannot be put to `ChannelError::WouldBlock` since embedded_io defines read/write to always block, there is no equivalent to `ChannelError::WouldBlock`. so this comment-line can be removed.

This probably indicates a deeper issue as LibOSDP expects all read/write to be non-blocking. We need to explore ReadReady/WriteReady traits and adjust the channel callbacks to handle the outcomes correctly.

Reply:
You are right. But currently we are only using `embedded_io::Error`, so that is mostly a problem implementers of Channel will have, not of `libosdp` directly.

This should probably be pointed out more clear in the doc comments of `Channel` though.

A quick google search revealed that none of the HALs I could find implement `ReadReady`/`WriteReady` currently. I think `WriteReady` is not stricly necessary, but `Read` alone will simply not work as expected. That is indeed problematic.

Nevertheless, I don't think this has to block this PR. It can be handled in a future PR.